mm2src/coins/eth.rs

```diff
@@ -3355,6 +3570,166 @@ fn rpc_event_handlers_for_eth_transport(ctx: &MmArc, ticker: String) -> Vec<RpcT
 #[inline]
 fn new_nonce_lock() -> Arc<AsyncMutex<()>> { Arc::new(AsyncMutex::new(())) }
 
+pub async fn eth_coin_from_conf_and_request_v2(
```

Comment:
This function could be a separate module

Reply:
Could you please refactor `eth_coin_from_conf_and_request_v2` and `eth_coin_from_conf_and_request` like it we did for UTXO coins:
https://github.com/KomodoPlatform/atomicDEX-API/blob/bb4ddf0831405ff4ea0e28888f9102e8e0468c82/mm2src/coins/utxo/utxo_builder/utxo_coin_builder.rs#L101-L116
So I suppose to add `EthCoinBuilder` trait:
```rust
trait EthCoinBuilder {
    type ResultCoin;
    type Error: NotMmError;

    async fn build(self) -> MmResult<Self::ResultCoin, Self::Error>;

    async fn build_eth_fields(&self) -> MmResult<EthCoinImpl, Self::Error> {
        let transport = self.build_transport().await?;
    }

    async fn build_transport(&self) -> MmResult<Web3<Web3Transport>, Self::Error>;
}
```
It's because there are some common parts within `eth_coin_from_conf_and_request` and `eth_coin_from_conf_and_request_v2`

Reply:
> This function could be a separate module

It's now in `v2_activation` module.

Reply:
> Could you please refactor `eth_coin_from_conf_and_request_v2` and `eth_coin_from_conf_and_request` like it we did for UTXO coins:
> 
> https://github.com/KomodoPlatform/atomicDEX-API/blob/bb4ddf0831405ff4ea0e28888f9102e8e0468c82/mm2src/coins/utxo/utxo_builder/utxo_coin_builder.rs#L101-L116
> 
> 
> So I suppose to add `EthCoinBuilder` trait:
> ```rust
> trait EthCoinBuilder {
>     type ResultCoin;
>     type Error: NotMmError;
> 
>     async fn build(self) -> MmResult<Self::ResultCoin, Self::Error>;
> 
>     async fn build_eth_fields(&self) -> MmResult<EthCoinImpl, Self::Error> {
>         let transport = self.build_transport().await?;
>     }
> 
>     async fn build_transport(&self) -> MmResult<Web3<Web3Transport>, Self::Error>;
> }
> ```
> 
> It's because there are some common parts within `eth_coin_from_conf_and_request` and `eth_coin_from_conf_and_request_v2`

This is actually requires a lot of refactoring including payload structs, should we do it in this PR ? :)

Reply:
ETH coin has less properties, and it's easier to implement such trait for ETH coin than to implement `UtxoCoinBuilder` for a UTXO coin, so I guess there shouldn't be a lot of refactoring

Reply:
What do you think @artemii235 ?

Reply:
I am waiting for Artem's answer for this refactoring.

Reply:
There are too many changes on this PR already. Also, UTXO coins have many variations like `QtumCoin`, `BchCoin` etc., for which using an additional builder abstraction is justified. I don't think that it's really required for ETH at this stage.